# Screening of microsporidia and relatives in genomes

This repository contains a **Snakemake**-based pipeline for genome binning and phylogenomic analysis, with a focus on intracellular parasites and other reduced-genome organisms. The pipeline is designed for Linux environments.

## Content

- [Pipeline Overview](#pipeline-overview)
- [Requirements](#Requirements)
- [Install](#install)
- [Running instructions](#running-instructions)
- [Examples](#examples)
- [Contact](#contact)


## Pipeline Overview

1. **Assembly downloading**

Downloads a reference genome assembly with a given NCBI identifier (e.g. `GCF_007210705.1`). Uses the NCBI FTP link to obtain the `.fna.gz` file, which is then unpacked into `input_reads/`.

2. **Artificial coverage**

Create coverage file required by `MaxBin2`. Each sequence is assigned an artificial coverage (default 300x), which is specified in `alignment/{id}_abundance.tsv`.

3. **Binning with MaxBin2**

Splits the assembly into individual bins based on coverage data. Results are saved to `maxbin_output/{id}/`.

4. **BUSCO Completeness Check**

Evaluates the completeness of each bin using BUSCO based on the fungi_odb10 lineage and optionally `microsporidia_odb10`. Results include `.faa` sequences of single-copy genes and completeness reports. Then automatically select the bin with the highest number of complete single-copy BUSCOs  based on BUSCO `.txt` reports.

7. **BUSCO Phylogenomics**

Uses the `BUSCO_phylogenomics.py` script and provided collection of microsporidia in `data_for_tree` archive to build a phylogenetic tree on best bins.

## System and memory requirements

- Linux (tested on Ubuntu 24.04.1, kernel 6.8 and Ubuntu 22.04.3, kernel 5.15)

- 60+ GB disk space for full analysis and >=32 GB RAM.

- Not supported: MacOS (FragGeneScan compatibility issues) / Windows (untested)

## Install

1. Clone this repository to your local machine:
```bash
git clone git@github.com:venikkus/microsporidia_screening.git
cd microsporidia_screening
```

2. Use the provided `environment.yaml` file to create the conda environment. If you don't have conda, use these [installation instuctions](https://www.anaconda.com/docs/getting-started/miniconda/install#linux):

```bash
conda env create -f environment.yaml    
conda activate screening
```

Clone `BUSCO_phylogenomics` tool from GitHub.
```bash
git clone git@github.com:jamiemcg/BUSCO_phylogenomics
```
or
```
git clone https://github.com/jamiemcg/BUSCO_phylogenomics.git
```

Then download and unzip the databases (this step is for people who need to use a VPN, since downloading is not available from ðŸ’…RussiaðŸ’…):
```bash
mkdir -p busco/fungi_odb10
mkdir -p busco/microsporidia_odb10

wget -c https://busco-data.ezlab.org/v5/data/lineages/microsporidia_odb10.2024-01-08.tar.gz -O busco/microsporidia_odb10/microsporidia_odb10.2024-01-08.tar.gz
wget -c https://busco-data.ezlab.org/v5/data/lineages/fungi_odb10.2024-01-08.tar.gz -O busco/fungi_odb10/fungi_odb10.2024-01-08.tar.gz

tar -xvzf busco/fungi_odb10.2024-01-08.tar.gz -C busco
tar -xvzf busco/microsporidia_odb10.2024-01-08.tar.gz -C busco
```

Download and unpack reference busco results archive from [link](https://mega.nz/file/Hvo3XIZb#AsMuHzxkmFBySby07WmKajqgCF8_kf7sKXjSa-cLtZs) manually or using [MEGA CMD](https://mega.io/ru/cmd#download):

```bash
tar -xzf data_for_tree.tar.gz
```

## Features

Output structure:
1. Downloaded assembly

`input_reads/{id}.fna`

Uncompressed FASTA file of the genome assembly.                                   

2. Abundance File

`alignment/{id}_abundance.tsv`

Simulated coverage values for MaxBin2.                                             

3. MaxBin Bins

`maxbin_output/{id}/{bin}.fasta`

Individual bins generated by MaxBin2.                                              

4. BUSCO Reports

`busco_output/{id}/run_{bin}/short_summary*.txt`
`busco_output/{id}/run_{bin}/full_table.tsv`

Completeness reports for each bin: percentage of S, D, F, and M BUSCO genes.      


5. Select best bin

`busco_output/{id}/best_fungi_bin.txt`

Name of the bin with the highest number of complete single-copy BUSCOs (S).       

6. Move to BUSCO Genes (selected)

`data_for_tree/busco_{id}.{bin}/run_fungi_odb10/busco_sequences/single_copy_busco_sequences/*.faa`

Single-copy BUSCO sequences extracted from the best bin.

7. Phylogenomics Results

`busco_phylo_results/`

Trees, alignments, and log files from the phylogenomic tree construction pipeline.

## Running instructions

You can run pipeline like this:
```bash
snakemake -p --cores [NUM_CORES] --use-conda --config id=[ASSEMBLY_ID] assembly=[ASSEMBLY_NAME]
```

- `[NUM_CORES]` â€” number of CPU cores to use (e.g., 4, 16, etc.)
- `[ASSEMBLY_ID]` â€” NCBI assembly accession (e.g., GCF_000001405.39)
- `[ASSEMBLY_NAME]` â€” name of the genome assembly (e.g., GRCh38.p13)

Then BUSCO pipeline is complete, run following command:
```bash
snakemake -s snakefile_phylo --cores [NUM_CORES] --config busco_input=[DATA_FOR_TREE]
```

## Examples

If you want quickly recreate ***part*** of this project:
```bash
chmod +x run_part.sh
./run_part.sh
```
Or add parameters manually.
```bash
snakemake --cores 50 --use-conda --config id=GCF_002237135.1 assembly=ASM223713v2
```

And then:
```bash
snakemake -s snakefile_phylo --cores 4 --config busco_input=data_for_tree
```

Or ***all*** project (it might take a lot of time).
```bash
chmod +x run_all.sh
./run_all.sh
```

And then:
```bash
snakemake -s snakefile_phylo --cores 4 --config busco_input=data_for_tree
```

If your process was accidentaly or purposely killed and marker folders was created, but not the required files, please delete marker folders and rerun.

## Contact

Please report any problems directly to the GitHub [issue tracker](https://github.com/venikkus/microsporidia_screening/issues).

Also, you can send your feedback to authors:
- [niksamusik@gmail.com](mailto:niksamusik@gmail.com)
- [poluzerov24@gmail.com](mailto:poluzerov24@gmail.com)