# Screening of microsporidia and relatives in genomes

This repository contains a **Snakemake**-based pipeline for genome binning and phylogenomic analysis, with a focus on intracellular parasites and other reduced-genome organisms. The pipeline is designed for Linux environments.

## Content

- [Pipeline Overview](#pipeline-overview)
- [Requirements](#Requirements)
- [Install](#install)
- [Running instructions](#running-instructions)
- [Examples](#examples)
- [Contact](#contact)


## Pipeline Overview

1. **Assembly downloading**

Downloads a reference genome assembly with a given NCBI identifier (e.g. `GCF_007210705.1`). Uses the NCBI FTP link to obtain the `.fna.gz` file, which is then unpacked into `input_reads/`.

2. **Fake coverage simulation**

Creates a fake coverage file required by `MaxBin2`. Each sequence is assigned an artificial coverage (default 300x), which is specified in `alignment/{id}_abundance.tsv`.

3. **Binning with MaxBin2**

Splits the assembly into individual bins based on coverage data. Results are saved to `maxbin_output/{id}/`.

4. **BUSCO Completeness Check**

Evaluates the completeness of each bin using BUSCO based on the fungi_odb10 lineage and optionally `microsporidia_odb10`. Results include `.faa` sequences of single-copy genes and completeness reports. Then automatically select the bin with the highest number of complete single-copy BUSCOs  based on BUSCO `.txt` reports.

7. **BUSCO Phylogenomics**

Uses the `BUSCO_phylogenomics.py` script to build a phylogenetic tree on best bins and provided collection of microsporidia.

## Requirements

- Linux operating system (MacOS is not supported due to FragGeneScan issues and pipeline wasn't tested on Windows), in our case: 

    >Linux version 6.8.0-55-generic (buildd@lcy02-amd64-095) (x86_64-linux-gnu-gcc-13 (Ubuntu 13.3.0-6ubuntu2~24.04) 13.3.0, GNU ld (GNU Binutils for Ubuntu) 2.42) #57-Ubuntu SMP PREEMPT_DYNAMIC
- Conda (You can download it from [here](https://www.anaconda.com/docs/getting-started/miniconda/install#quickstart-install-instructions)), in our case:
    >conda 24.11.3


## Install

1. Clone this repository to your local machine:
```bash
git clone git@github.com:venikkus/microsporidia_screening.git
cd microsporidia_screening
```

2. Use the provided `environment.yaml` file to create the conda environment:

```bash
conda env create -f environment.yaml    
conda activate screening
```

Clone `BUSCO_phylogenomics` tool from GitHub.
```bash
git clone https://github.com/jamiemcg/BUSCO_phylogenomics
```

Then download and unzip the databases (this step for people, who have a server and need to use VPN, if work from ðŸ’…RussiaðŸ’…):
```bash
mkdir -p busco/fungi_odb10
mkdir -p busco/microsporidia_odb10

wget -c https://busco-data.ezlab.org/v5/data/lineages/microsporidia_odb10.2024-01-08.tar.gz -O busco/microsporidia_odb10/microsporidia_odb10.2024-01-08.tar.gz
wget -c https://busco-data.ezlab.org/v5/data/lineages/fungi_odb10.2024-01-08.tar.gz -O busco/fungi_odb10/fungi_odb10.2024-01-08.tar.gz

tar -xvzf busco/fungi_odb10.2024-01-08.tar.gz -C busco
tar -xvzf busco/microsporidia_odb10.2024-01-08.tar.gz -C busco
```
## Features

Output structure:
1. Downloaded assembly

`input_reads/{id}.fna`

Uncompressed FASTA file of the genome assembly.                                   

2. Abundance File

`alignment/{id}_abundance.tsv`

Simulated coverage values for MaxBin2.                                             

3. MaxBin Bins

`maxbin_output/{id}/{bin}.fasta`

Individual bins generated by MaxBin2.                                              

4. BUSCO Reports

`busco_output/{id}/run_{bin}/short_summary*.txt`
`busco_output/{id}/run_{bin}/full_table.tsv`

Completeness reports for each bin: percentage of S, D, F, and M BUSCO genes.      


5. Select best bin

`busco_output/{id}/best_fungi_bin.txt`

Name of the bin with the highest number of complete single-copy BUSCOs (S).       

6. Move to BUSCO Genes (selected)

`data_for_tree/busco_{id}.{bin}/run_fungi_odb10/busco_sequences/single_copy_busco_sequences/*.faa`

Single-copy BUSCO sequences extracted from the best bin.

7. Phylogenomics Results

`busco_phylo_results/`

Trees, alignments, and log files from the phylogenomic tree construction pipeline.

## Running instructions

You can run pipeline like this:
```bash
snakemake -p --cores [NUM_CORES] --use-conda --config id=[ASSEMBLY_ID] assembly=[ASSEMBLY_NAME]
```

- `[NUM_CORES]` â€” number of CPU cores to use (e.g., 4, 16, etc.)
- `[ASSEMBLY_ID]` â€” NCBI assembly accession (e.g., GCF_000001405.39)
- `[ASSEMBLY_NAME]` â€” name of the genome assembly (e.g., GRCh38.p13)

And then BUSCO pipeline is complete, run following command:
```bash
snakemake -s snakefile_phylo --cores [NUM_CORES] --config busco_input=[DATA_FOR_TREE]
```

## Examples

If you want quickly recreate ***part*** of this project:
```bash
chmod +x run_part.sh
./run_part.sh
```
Or add parameters manually.
```bash
snakemake --cores 50 --use-conda --config id=GCF_002237135.1 assembly=ASM223713v2
```

Or ***all*** project (it might take a lot of time).
```bash
chmod +x run_all.sh
./run_all.sh
```

If your process was accidentaly or purposely killed and marker folders was created, but not the required files, please delete marker folders and rerun.

## Contact

Please report any problems directly to the GitHub
[issue tracker](https://github.com/venikkus/microsporidia_screening/issues).<br/>
Also, you can send your feedback to
[niksamusik@gmail.com](mailto:niksamusik@gmail.com) or [
poluzerov24@gmail.com](mailto:poluzerov24@gmail.com).
